examples/log-source-app/log-source-common/DiagnosticLogsCommandHandler.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the Liceunse at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <log-source-common/DiagnosticLogsCommandHandler.h>
+
+CHIP_ERROR DiagnosticLogsCommandHandler::PushLog(const chip::ByteSpan & payload)
```

Comment:
Is there a reason it is implemented in the example app instead in the common `src/app` module? Even the logging redirection is platform-agnostic, so I'm wondering if most of this functionality couldn't be implemented in some generic (although configurable) way?

Reply:
Unlike other clusters, the logs source/sink doesn't fit the server/client model. The controller (aka client), can either by source or sink. So I'm not sure it is feasible to put it in `src/app/clusters`